### ISSUE TITLE
Feature request: Allow specifying uuid on create

### ISSUE THREAD

**Issue by Joshfindit** [neo4jrb/neo4j#1141]
In some cases, the UUIDs for some nodes already exists in a dataset so when importing we'd like to carry those uuids across.

    node = Node.create
    node.name = "Name"
    node.uuid = "dc1be2f2-1d08-475d-8f55-8aec8c50dce6"
    node.save

Even a simple "is this a uuid, and if so, use it instead of generating" could work here.

**Comment by cheerfulstoic** [comment]
Yeah, this is something that we should perhaps figure out.  Maybe it could be that we allow disabling this check temporarily.  You might be able to get something to work by specifying `id_property` manually and defining an `auto` method which reads the value from an instance variable which you set before-hand.  Ugly, but it could be a workaround for now

**Comment by bloomdido** [comment]
+1

**Comment by Joshfindit** [comment]
Any further thoughts on this?
I'm about to step in to the phase when I'm testing with live data, and it's going to be crucial to be able to keep the UUID across db wipes.

**Comment by subvertallchris** [comment]
This is a request that's popped up pretty consistently for a while, so if it's something we all agree needs to happen, I'm happy to set aside some time in the coming weeks to work on it. In the meantime, like Brian said, you can work around it by define a custom `id_property` method.

```ruby
class MyModel
  include Neo4j::ActiveNode
  attr_writer :uuid_placeholder
  id_property :uuid, on: :my_uuid_method

  def my_uuid_method
    @uuid_placeholder || SecureRandom.uuid
  end
end

# Set UUID
MyModel.create(uuid_placeholder: 'myuuid')

# Allow UUID to be created automatically
MyModel.create
```

**Comment by bloomdido** [comment]
I'm probably going to fork and fix this myself in the next day or two, along with a few other things I need to get working for my project. So if nobody gets to it sooner than that, I'll look into adding a PR for whatever I come up with.

**Comment by subvertallchris** [comment]
Very cool. IMO, the easiest fix for this would be to have `id_property` call `property` at some point to actually register the property like any other one declared in the model. We have a few workarounds to deal with the fact that this doesn't happen, so it should lead to simpler code elsewhere.

**Comment by Joshfindit** [comment]
@subvertallchris: Quick question on the workaround; what would the syntax be to use `uuid_placeholder` with `Class.merge`?

**Comment by Joshfindit** [comment]
Coming back to this; the quickest way that I can find to deal with `Merge` when using the `uuid_placeholder`workaround is to do something like this:

``` ruby
if MyModel.find(importedUUID)
  MyModel.merge(id: importedUUID)
else
  MyModel.create (uuid_placeholder: importedUUID)
end
```

**Comment by Joshfindit** [comment]
Any recent thoughts on this request?

**Comment by cheerfulstoic** [comment]
Sorry for the delay.  Unfortunately no progress on this right now.  The maintainers have been busy with various things and haven't had much time for the gems aside from answering questions

**Comment by Joshfindit** [comment]
As a side note for other people who want this functionality; Instead of `uuid_placeholder`, I run raw cypher to create the node with the specific uuid (and proper labels), then do `MyModel.find` to fill in the properties/relationships.

This has helped tremendously with database seeding, backup, and restore while having minimal impact on app usage.
